### ISSUE TITLE
[BUG] The total GPU memory is constantly changing for Apple M1

### ISSUE THREAD

**Issue by aiwantaozi** [fastfetch-cli/fastfetch#1273]
# General description of bug:

* What happened:
1. Running fastfetch with command in MacOS 14.5 M1 with command `fastfetch --gpu-temp true --gpu-detection-method pci --gpu-driver-specific true --config only-gpu.jsonc --format json`
```json
{
    "$schema": "https://github.com/fastfetch-cli/fastfetch/raw/dev/doc/json_schema.json",
    "modules": ["gpu"]
}
```

2. Got the result
```
[
  {
    "type": "GPU",
    "result": [
      {
        "coreCount": 16,
        "coreUsage": 37.0,
        "memory": {
          "dedicated": {
            "total": null,
            "used": null
          },
          "shared": {
            "total": 9698246656,
            "used": 422166528
          }
        },
        "driver": "com.apple.AGXG13X",
        "name": "Apple M1 Pro",
        "temperature": 48.058441162109375,
        "type": "Integrated",
        "vendor": "Apple",
        "platformApi": "Metal 3",
        "frequency": 1296,
        "deviceId": 4294969775
      }
    ]
  }
]
```
3. Ran the same command again, the memory.shared.total is changed.
```json
[
  {
    "type": "GPU",
    "result": [
      {
        "coreCount": 16,
        "coreUsage": 30.0,
        "memory": {
          "dedicated": {
            "total": null,
            "used": null
          },
          "shared": {
            "total": 9625436160,
            "used": 308363264
          }
        },
        "driver": "com.apple.AGXG13X",
        "name": "Apple M1 Pro",
        "temperature": null,
        "type": "Integrated",
        "vendor": "Apple",
        "platformApi": "Metal 3",
        "frequency": 1296,
        "deviceId": 4294969775
      }
    ]
  }
]
```

4. After stopping the application used GPU, then the total changed greatly
```json
[
  {
    "type": "GPU",
    "result": [
      {
        "coreCount": 16,
        "coreUsage": 54.0,
        "memory": {
          "dedicated": {
            "total": null,
            "used": null
          },
          "shared": {
            "total": 3366600704,
            "used": 314097664
          }
        },
        "driver": "com.apple.AGXG13X",
        "name": "Apple M1 Pro",
        "temperature": 49.04446220397949,
        "type": "Integrated",
        "vendor": "Apple",
        "platformApi": "Metal 3",
        "frequency": 1296,
        "deviceId": 4294969775
      }
    ]
  }
]
```

5. Found after this [commit](https://github.com/fastfetch-cli/fastfetch/commit/c549a5cab8b362219fbbcb6048346f74d0a23567#diff-3cd704dc84802940fd6219278e7c2de25c1533038ea9b1720d0f21246f40a1d2), the [recommendedMaxWorkingSetSize](https://developer.apple.com/documentation/metal/mtldevice/2369280-recommendedmaxworkingsetsize?language=objc) is removed.

6. Use fastfetch 2.20.0 and run the same command, the shared total is consistent across multiple runs.
```json
[
  {
    "type": "GPU",
    "result": [
      {
        "coreCount": 16,
        "memory": {
          "dedicated": {
            "total": null,
            "used": null
          },
          "shared": {
            "total": 22906503168,
            "used": 65536
          }
        },
        "driver": "com.apple.AGXG13X",
        "name": "Apple M1 Pro",
        "temperature": 47.71730041503906,
        "type": "Integrated",
        "vendor": "Apple",
        "platformApi": "Metal 3",
        "frequency": 1296.0,
        "deviceId": 4294969775
      }
    ]
  }
]
```

* What should happen:
Can recommendedMaxWorkingSetSize be used to represent shared total memory in Mac UMA architecture?

* Fastfetch version used: fastfetch 2.24.0 (aarch64)
* Did it work in an older version: fastfetch 2.20.0 (aarch64)
* Where did you get the binary: GitHub release
* Does this issue still occur in [the latest dev build](https://github.com/fastfetch-cli/fastfetch/actions/)? yes

**Comment by CarterLi** [comment]
> The total GPU memory is constantly changing for Apple M1

For UMA, VMEM is always dynamically allocated, isn't it?

> the [recommendedMaxWorkingSetSize](https://developer.apple.com/documentation/metal/mtldevice/2369280-recommendedmaxworkingsetsize?language=objc) is removed

Because I found that the value never changes, consistantly 10.7 GiB on my 16 GiB M1 Pro, even if the free memory is less than 10 GiB. In addition, the used memory size `65536` doesn't make sense at all.

**Comment by aiwantaozi** [comment]
Based on the Apple Tech Talks, the value of recommendedMaxWorkingSetSize can be queried from the Metal API but remains constant throughout execution: https://developer.apple.com/videos/play/tech-talks/10580/?time=424.

The currentAllocatedSize, on the other hand, appears to represent only the VRAM allocated by the current application, not the entire GPU's memory usage. I tested this by continually allocating memory in application A and observed that currentAllocatedSize increased in application A. However, when I queried currentAllocatedSize from application B, it consistently returned 65535.

To get the total VRAM size of the GPU, I think we could use recommendedMaxWorkingSetSize, and for the VRAM currently in use, retrieve data from PerformanceStatistics for the whole GPU.
